### PR TITLE
feat: #394 add errorElement to Route

### PR DESCRIPTION
**Pull request by AKiroi**
Need to test!

### CODE REVIEW

File: mw-webapp/src/App.tsx [hunk 1]
```diff
@@ -1,82 +1,30 @@
...
-
-    if (getIsHomePage()) {
-      navigate(defaultPagePath);
-    }
-  };
-
-  useEffect(() => {
-    AuthService.listenAuthStateChange({onLogIn, onLogOut});
-  }, []);
+  const [isInitialization, setIsInitialization] = useState(true);
```
Comment (tritonJS826) [inline mw-webapp/src/App.tsx:15]: isInitialized, setIsInitialized. And let's make it false by default

File: mw-webapp/src/GlobalContext.ts [hunk 2]
```diff
@@ -8,6 +8,13 @@ const DEFAULT_SET_USER = () => {
   throw Error("The user context is not initialized properly");
 };
 
+/**
+ * Default setIsInitialization
+ */
+const DEFAULT_SET_INITIALIZATION = () => {
+  throw Error("The app is not initialization");
```
Comment (tritonJS826) [inline mw-webapp/src/GlobalContext.ts:15]: Let's s use more specific message, like "This is stub function for the initialization. This function should not be called"

File: mw-webapp/src/logic/page404/Page404.tsx [hunk 3]
```diff
@@ -7,6 +10,37 @@ const ERROR_404 = "404 NOT FOUND";
  * Page 404 will be displayed if app's address is not correct
  */
 export const Page404 = () => {
+  const error = useRouteError();
+  if (isRouteErrorResponse(error)) {
+    return (
+      <>
```
Comment (tritonJS826) [inline mw-webapp/src/logic/page404/Page404.tsx:16]: Unnecessary fragment

File: mw-webapp/src/logic/wayPage/WayPage.tsx [hunk 4]
```diff
@@ -118,6 +118,7 @@ interface WayPageProps {
  * Way page
  */
 export const WayPage = (props: WayPageProps) => {
+  // Throw new Error("waypage error");
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/WayPage.tsx:121]: !

**Review by tritonJS826 — APPROVED** [review]

**Comment by tritonJS826** [comment]
Yeah, that's nice, you can merge it after fixing these minor things.

File: mw-webapp/src/App.tsx (hunk 1 above)
Comment (AKiroi) [reply mw-webapp/src/App.tsx:15]: fixed

File: mw-webapp/src/GlobalContext.ts (hunk 2 above)
Comment (AKiroi) [reply mw-webapp/src/GlobalContext.ts:15]: fixed

File: mw-webapp/src/logic/page404/Page404.tsx (hunk 3 above)
Comment (AKiroi) [reply mw-webapp/src/logic/page404/Page404.tsx:16]: done

File: mw-webapp/src/logic/wayPage/WayPage.tsx (hunk 4 above)
Comment (AKiroi) [reply mw-webapp/src/logic/wayPage/WayPage.tsx:121]: fixed

**Review by tritonJS826 — APPROVED** [review]

**Pull request merged**